compiler/rustc_session/src/filesearch.rs

```diff
@@ -164,6 +164,22 @@ pub fn get_or_default_sysroot() -> Result<PathBuf, String> {
         fix_windows_verbatim_for_gcc(&path)
     }
 
+    // Use env::current_exe() to get the path of the executable following
+    // symlinks/canonicalizing components.
+    #[cfg(target_os = "linux")]
+    fn default_from_current_exe() -> Result<PathBuf, std::io::Error> {
```

Comment:
This breaks the case that https://github.com/rust-lang/rust/pull/103660 was supposed to fix (auto detection of an external tool depending on rustc_driver). Maybe instead check if the parent dir of `librustc_driver.so` is not `lib` and in that case pop two directories instead of one.

Reply:
Thank you for the suggestion! I have changed the logic to address this comment.